src/keys/FileKey.h

```diff
@@ -28,10 +28,19 @@ class QIODevice;
 class FileKey: public Key
 {
 public:
+    enum Type {
```

Comment:
How would you feel about using a scoped enum here? Their usage may be more verbose, e.g., `FileKey::Type::HashFileKeyed` instead of `FileKey::Hashed`, but they are safer as they prevent implicit conversion to their underlying integral type in comparison operations.

Reply:
I don't think the additional namespace encapsulation would remedy the additional typing overhead here. For a larger class with more members and possibly different enum types, I would totally agree with using class enums, but here I don't think it'd have any benefit over traditional eums.

Reply:
Ok.

Reply:
It's not just the additional namespace encapsulation though; the safer part comes from preventing implicit type conversion.

Reply:
True. But what would you compare these enums with that would cause problems?

Reply:
In practice, it can be an integral type, such as an int.

Reply:
Sure, but what would be the problem here? Where would you compare it with something so that automatic type conversion could fuck you up? And I don't mean in general (yes, in general that's a bad thing). I mean in this specific case.

Reply:
I get your point, in this particular case, and because code reviews are practiced, it would be difficult for such an issue to occur.